src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAITextToImageClient.cs

```diff
@@ -0,0 +1,207 @@
+                // For now, we only support DataContent for image editing as OpenAI's API expects image data in a stream.
+                Throw.ArgumentException(
+                    "The original image must be a DataContent instance containing image data.",
+                    nameof(originalImages));
+            }
+        }
+
+        GeneratedImageCollection result = await _imageClient.GenerateImageEditsAsync(
+            imageStream, fileName, prompt, options?.Count ?? 1, openAIOptions, cancellationToken).ConfigureAwait(false);
+
+        return ToTextToImageResponse(result);
+    }
+
+    /// <inheritdoc />
+#pragma warning disable S1067 // Expressions should not be too complex
+    public object? GetService(Type serviceType, object? serviceKey = null) =>
+        serviceType is null ? throw new ArgumentNullException(nameof(serviceType)) :
+        serviceKey is not null ? null :
+        serviceType == typeof(TextToImageClientMetadata) ? _metadata :
+        serviceType == typeof(ImageClient) ? _imageClient :
+        serviceType.IsInstanceOfType(this) ? this :
+        null;
+#pragma warning restore S1067 // Expressions should not be too complex
+
+    /// <inheritdoc />
+    void IDisposable.Dispose()
+    {
+        // Nothing to dispose. Implementation required for the ITextToImageClient interface.
+    }
+
+    /// <summary>
+    /// Converts a <see cref="Size"/> to an OpenAI <see cref="GeneratedImageSize"/>.
+    /// </summary>
+    /// <param name="requestedSize">User's requested size.</param>
+    /// <returns>Closest supported size.</returns>
+    private static GeneratedImageSize? ToOpenAIImageSize(Size? requestedSize) =>
+        requestedSize is null ? null : new GeneratedImageSize(requestedSize.Value.Width, requestedSize.Value.Height);
+
+    /// <summary>Converts a <see cref="GeneratedImageCollection"/> to a <see cref="TextToImageResponse"/>.</summary>
+    private static TextToImageResponse ToTextToImageResponse(GeneratedImageCollection generatedImages)
+    {
+        string contentType = "image/png"; // Default content type for images
+
+        // OpenAI doesn't expose the content type, so we need to read from the internal JSON representation.
+        // https://github.com/openai/openai-dotnet/issues/561
+        IDictionary<string, BinaryData>? additionalRawData = typeof(GeneratedImageCollection)
+            .GetProperty("SerializedAdditionalRawData", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+            ?.GetValue(generatedImages) as IDictionary<string, BinaryData>;
+
+        if (additionalRawData?.TryGetValue("output_format", out var outputFormat) ?? false)
+        {
+#pragma warning disable IL2026, IL3050 // Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code
+            contentType = $"image/{outputFormat.ToObjectFromJson<string>()}";
```

Comment:
We should avoid doing anything that needs these linker/NativeAOT suppressions. Here, you should be able to use the overload of ToObjectFromJson that takes a JsonTypeInfo, and call it with `(JsonTypeInfo<string>)AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(string))`

Reply:
I was banging my head against this one - it's known to be a Json string, which is just a quoted string encoded as Utf8 - I thought I was safe to suppress this since string would always be supported.

Thanks for the pointer - I think this will work.